wizard.sh

```diff
@@ -0,0 +1,299 @@
+#!/bin/bash
+
+set -e
+
+untracked="(untracked)"
+REPO=$(cd $(dirname $0); pwd)
+LINT="false"
+BUILD="false"
+PUSH_LATEST="false"
+RELEASE=""
+
+debugInfo () {
+  echo "Repo:                     $REPO"
+  echo "Will lint:                $LINT"
+  echo "Will build:               $BUILD"
+  echo "Will release:             $RELEASE"
+  echo "Will push latest docker:  $PUSH_LATEST"
+  echo "Use Docker:               $USE_DOCKER"
+  echo "Is CI:                    $CI"
+}
+
+dockerLogin () {
+  gpg --batch --gen-key <<-EOF
+%echo Generating a standard key
+Key-Type: DSA
+Key-Length: 1024
+Subkey-Type: ELG-E
+Subkey-Length: 1024
+Name-Real: Meshuggah Rocks
+Name-Email: meshuggah@example.com
+Expire-Date: 0
+# Do a commit here, so that we can later print "done" :-)
+%commit
+%echo done
+EOF
+
+  key=$(gpg --no-auto-check-trustdb --list-secret-keys | grep ^sec | cut -d/ -f2 | cut -d" " -f1)
+  pass init $key
+
+  if [ "$(command -v docker-credential-pass)" = "" ]; then
+    docker run --rm -itv /usr/local/bin:/src filebrowser/dev sh -c "cp /go/bin/docker-credential-pass /src"
+  fi
+
+  echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
+}
+
+dockerPushLatest () {
+  docker build -t filebrowser/filebrowser .
+  dockerLogin
+  docker push filebrowser/filebrowser
+  docker logout
+}
+
+installRice () {
+  if ! [ -x "$(command -v rice)" ]; then
+    go get github.com/GeertJohan/go.rice/rice
+  fi
+}
+
+buildAssets () {
+  installRice
+  cd $REPO/frontend
+
+  if [ -d "dist" ]; then
+    rm -rf dist/*
+  fi;
+
+  yarn install
+  yarn build
+
+  cd $REPO/http
+  rice embed-go
+}
+
+updateVersion () {
+  from=$1
+  to=$2
+
+  echo "Updating version from \"$from\" to \"$to\""
+  sed -i.bak "s|$from|$to|g" $REPO/types/version.go
+}
+
+buildBinary () {
+  cd $REPO
+  go get -v ./...
+  updateVersion $untracked "($COMMIT_SHA)"
+  echo "Build CLI"
+  go build -a -o filebrowser
```

Comment:
In `filebrowser/dev` envvar `CGO_ENABLED=0` is set. See https://github.com/filebrowser/docker-dev/blob/master/Dockerfile#L12. Thus, this script will produce a portable binary when executed inside `filebrowser/dev`. Otherwise, the binary will be static but it will depend on the standard library available on the system.

Therefore, it might be desirable to prefix line 88 with `CGO_ENABLED=0`.

Reply:
Isn't CGO disabled by default though? We can absolutely add it anyway eheh

Reply:
Added!

Reply:
It is enabled by default for native builds, and disabled for cross-compilation. See: https://golang.org/cmd/cgo/

> The cgo tool is enabled by default for native builds on systems where it is expected to work. It is disabled by default when cross-compiling. You can control this by setting the CGO_ENABLED environment variable when running the go tool: set it to 1 to enable the use of cgo, and to 0 to disable it.